src/transformers/models/pop2piano/modeling_pop2piano.py

```diff
@@ -0,0 +1,1375 @@
+        )
+
+        encoder_config = copy.deepcopy(config)
+        encoder_config.is_decoder = False
+        encoder_config.use_cache = False
+        encoder_config.is_encoder_decoder = False
+
+        self.encoder = Pop2PianoStack(encoder_config, self.shared)
+
+        decoder_config = copy.deepcopy(config)
+        decoder_config.is_decoder = True
+        decoder_config.is_encoder_decoder = False
+        decoder_config.num_layers = config.num_decoder_layers
+        self.decoder = Pop2PianoStack(decoder_config, self.shared)
+
+        self.lm_head = nn.Linear(config.d_model, config.vocab_size, bias=False)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+        # Model parallel
+        self.model_parallel = False
+        self.device_map = None
+
+    def get_input_embeddings(self):
+        return self.shared
+
+    def set_input_embeddings(self, new_embeddings):
+        self.shared = new_embeddings
+        self.encoder.set_input_embeddings(new_embeddings)
+        self.decoder.set_input_embeddings(new_embeddings)
+
+    def set_output_embeddings(self, new_embeddings):
+        self.lm_head = new_embeddings
+
+    def get_output_embeddings(self):
+        return self.lm_head
+
+    def get_encoder(self):
+        return self.encoder
+
+    def get_decoder(self):
+        return self.decoder
+
+    def get_mel_conditioner_outputs(self, inputs_embeds, composer, generation_config, attention_mask=None):
+        composer_to_feature_token = generation_config.composer_to_feature_token
+        if composer not in composer_to_feature_token.keys():
+            raise ValueError(
+                f"Please choose a composer from {list(composer_to_feature_token.keys())}. Composer received - {composer}"
+            )
+        composer_value = composer_to_feature_token[composer]
+        composer_value = torch.tensor(composer_value, device=self.device)
+        composer_value = composer_value.repeat(inputs_embeds.shape[0])
+
+        embedding_offset = min(composer_to_feature_token.values())
+
+        inputs_embeds = self.mel_conditioner(
+            feature=inputs_embeds,
+            index_value=composer_value,
+            embedding_offset=embedding_offset,
+        )
+        if attention_mask is not None:
+            # check for zero pads (ion between two examples)
+            zero_pads = [e for e, i in enumerate(attention_mask) if i.max() == 0]
+            inputs_embeds[zero_pads] = 0.0
+
+            # since self.mel_conditioner adds a new array at the front of inputs_embeds we need to do the same for attention_mask to keep the shapes same
+            front_ones = torch.ones([attention_mask.shape[0], 1])
+            attention_mask = torch.concatenate([front_ones, attention_mask], axis=1)
+            attention_mask[zero_pads] = 0
+
+        else:
+            attention_mask = torch.ones([inputs_embeds.shape[0], inputs_embeds.shape[1]], dtype=torch.long)
```

Comment:
I don't think we need this - we can just return `attention_mask=None` if it's `None`. The modelling code will handle a `None` attention mask for us